packages/promise-tracker/lib/index.ts

```diff
@@ -0,0 +1,104 @@
+const _outstandingPromiseInstanceMap = new WeakMap<Object, Set<Promise<any>>>();
+const _outstandingPromiseMap = new Map<Promise<any>, boolean>();
+
+export function tracked<T extends Object>(
+  target: T,
+  key: string | Symbol,
+  descriptor: PropertyDescriptor
+) {
+  if (typeof descriptor.value === "function") {
+    const impl = descriptor.value;
+
+    // wrapper function that keeps a list of outstanding promises for each
+    // instance of DashboardMessageBusClient
+    function newImpl(...args: any[]) {
+      const returnValue = impl.apply(this, args);
+
+      // only track async functions
+      if (returnValue.then && typeof returnValue.then === "function") {
```

Comment:
Can `returnValue` be `null` or `undefined` here? If so, this will throw.

Reply:
good catch - will add a test for this and fix it

Reply:
done